### PR TITLE
Added examples for the common use cases when using the client

### PR DESCRIPTION
**Pull request by gabriela-yoncheva**
- Modify/Create a feature in a thing (with a certain definition)
- Modify a feature's property
- Messages request-response handling flow

Signed-off-by: Gabriela Yoncheva <fixed-term.Gabriela.Yoncheva@bosch.io>

### CODE REVIEW

**Comment by gabriela-yoncheva** [comment]
@thjaeckle Please do not merge this PR until it is approved by a member of the team.

File: examples/working_with_features.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    # Send the Ditto command.
+    envelope = command.envelope(response_required=False)
+
+    client.send(envelope)  
+
+# Test feature modifiation actions
+create_or_modify_feature(client)
+create_or_modify_feature_property(client)
+delete_feature(client)
+delete_feature_property(client)
```
Comment (konstantina-gramatova) [inline examples/working_with_features.py:74]: All these commands must be performed after the client has notified it is conneced (i.e. in the on_connect hook). Currently, there is no guarantee that they will be executed properly. Please, fix it.


Another hint:
To keep the test representation "clean" - i.e. no changes after executing the example, maybe a better idea is to order the calls like - create/modify feature, create/modify a feature's propery, delete property, delete feature.

File: examples/working_with_features.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+client.connect("localhost")
+
+def create_or_modify_feature(client):
+       
+    # Define the feature to be created or how you want it to be after modification
+    # You can provide a semantic definition of your feature
+    myFeature = Feature().with_definition_from("my.model.namespace:FeatureModel:1.0.0").with_property("myProperty", "myValue")
+    
+    # Create your Ditto command. Modify acts as an upsert - it either updates or creates features.
+    command = Command(NamespacedID().from_string("test.ns:test-name")).feature("myFeatureID").twin().modify(myFeature)
```
Comment (konstantina-gramatova) [inline examples/working_with_features.py:33]: It's a good practice to name the features in PascalCase. Please, fix it.

File: examples/working_with_features.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+client = Client()
+client.connect("localhost")
+
+def create_or_modify_feature(client):
+       
+    # Define the feature to be created or how you want it to be after modification
+    # You can provide a semantic definition of your feature
+    myFeature = Feature().with_definition_from("my.model.namespace:FeatureModel:1.0.0").with_property("myProperty", "myValue")
```
Comment (konstantina-gramatova) [inline examples/working_with_features.py:30]: Think about having the commonly used strings as private constants defined for the example - test namespaces/definition IDs, entity name, feature ID, etc.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

File: examples/message_request_response_handling.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+property_id = "myProperty"
+definition_id = DefinitionID().from_string("my.model.namespace:FeatureModel:1.0.0")
+my_feature = Feature().with_definition(definition_id).with_property(property_id, "myValue")
+req_topic = "command///req/" + str(thing_id) + "/"
+message_subject = "some-command"
+
+
+def send_inbox_message():
+    live_message = Message(thing_id).inbox(subject=message_subject).with_payload("some_payload")
+    live_message_envelope = live_message.envelope(response_required=True)
```
Comment (tinchouuu) [inline examples/message_request_response_handling.py:36]: Correlation ID is not set - please, add it.

File: examples/message_request_response_handling.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+feature_id = "MyFeatureID"
+property_id = "myProperty"
+definition_id = DefinitionID().from_string("my.model.namespace:FeatureModel:1.0.0")
+my_feature = Feature().with_definition(definition_id).with_property(property_id, "myValue")
+req_topic = "command///req/" + str(thing_id) + "/"
+message_subject = "some-command"
+
+
+def send_inbox_message():
+    live_message = Message(thing_id).inbox(subject=message_subject).with_payload("some_payload")
```
Comment (tinchouuu) [inline examples/message_request_response_handling.py:35]: Should be possible to use the short notation for the .inbox(message_subject) call here.

**Review by tinchouuu — CHANGES_REQUESTED** [review]

File: examples/message_request_response_handling.py (hunk 4 above)
Comment (konstantina-gramatova) [inline examples/message_request_response_handling.py:36]: Correlation ID is missing - please, set it in the example.

File: examples/message_request_response_handling.py (hunk 5 above)
Comment (konstantina-gramatova) [inline examples/message_request_response_handling.py:35]: The short notation of the .inbox(message_subject) should be possible to be used as well.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

**Review by konstantina-gramatova — APPROVED** [review]
LGTM

**Pull request merged**